stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Attachment.kt

```diff
@@ -42,7 +49,10 @@ public data class Attachment(
 
     @IgnoreSerialisation
     @IgnoreDeserialisation
-    override var extraData: MutableMap<String, Any> = mutableMapOf()
+    override var extraData: MutableMap<String, Any> = mutableMapOf(),
+
+    @SerializedName("upload_complete")
+    var uploadComplete: Boolean? = null
```

Comment:
I'm using `uploadComplete` instead of serializing `uploadState` as a sealed class would require soe work in the Gson library... But I'm open to other options, we can make this a enum and serialise it... or make Gson serialiase a it as sealed class too.  What do you (whoever is reviewing) think?

Reply:
If you want to add some custom fields to model there is `extraData` field. As I understood API convention it's the main reason for `extraData` existing. And your field could be backed field, something like:
```kotlin

class Attachment {
...
var uploadCompleted: Boolean?
 get() =
  if (extraData.contains("uploadCompleted")) 
       extraData.get("uploadCompleted") as Boolead
   else 
       null
  
 set(value: Boolean) = extraData.put("uploadCompleted", value)
}
```
@JcMinarro Could you prove my point?

Reply:
That was my first option, but we ask to `extraData` don't be serialised in this class. 

```
@IgnoreSerialisation
    @IgnoreDeserialisation
    override var extraData: MutableMap<String, Any> = mutableMapOf(),
```

I can also change extraData and make it serialisable, I just wonder why was it make this way and maybe it shouldn't be changed. 🤔

Reply:
Indeed, it's strange... I saw conversations that other teams definitely uses this to store custom attributes

Reply:
Yep... overall I think that that using the extraData looks a better solution, instead of creating one more field. I don't see a problem in sending it do backend, I can be useful for future features